infrastructure/flake.nix

```diff
@@ -0,0 +1,71 @@
+{
+  description = "Pocket Size Fund development shell";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+  }:
+    flake-utils.lib.eachDefaultSystem (
+      system: let
+        pkgs = import nixpkgs {
+          inherit system;
+        };
+        pythonEnv = pkgs.python312.withPackages (ps:
```

Comment:
Mismatching Python versions - should we do 3.13 or 3.12 across the repository? I also get that some services might use different versions so just clearing that up.